ipld/kamt/src/pointer.rs

```diff
@@ -293,3 +247,67 @@ fn unsplit_ext(
 
     Ok(ext)
 }
+
+#[cfg(test)]
+mod test {
+    use std::collections::BTreeMap;
+    use std::fmt::Debug;
+
+    use fvm_ipld_encoding::de::DeserializeOwned;
+    use fvm_ipld_encoding::{from_slice, to_vec, BytesDe};
+    use serde::Serialize;
+
+    use crate::ext::Extension;
+    use crate::pointer::Pointer;
+    use crate::KeyValuePair;
+
+    fn check_encoding<T: DeserializeOwned, V: Serialize>(expected: &T, input: &V)
+    where
+        T: Debug + Eq,
+    {
+        let decoded: T = from_slice(&to_vec(&input).unwrap()).unwrap();
+        assert_eq!(expected, &decoded);
```

Comment:
Shouldn't it check the other way around as well?

Reply:
I've changed this to check that both types converge to the same encoding. That should be safer.